Refactor library folder API logic into FoldersService

davelopez opened the pull request:
## What did you do? 
- Add basic API tests for the library folders API.
- Removed unused mixins (`UsesLibraryMixin`, `UsesLibraryMixinItems`) from `FoldersController`.
- Move the `FoldersController` logic into the `FoldersManager` class to be reused by the future FastAPI controller.


## Why did you make this change?
This will simplify the migration to FastAPI of this route in a future PR as well as provide some automated tests as a harness.


## How to test the changes? 
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).

davelopez committed:
Remove unused UsesLibraryMixin

davelopez committed:
Add basic API tests for library folders

davelopez committed:
Add a couple more tests for expected errors

davelopez committed:
Refactor controller logic into FoldersManager

jmchilton commented on lib/galaxy/managers/folders.py:
```diff
@@ -309,3 +312,257 @@ def cut_and_decode(self, trans, encoded_folder_id):
         return self.decode_folder_id(trans, self.cut_the_prefix(encoded_folder_id))
+
+
+class FoldersManager:
```
What is the difference between the FolderManager and FoldersManager? Should these be the same class?

davelopez replied:
The `s` character... just kidding 😆 

The idea is that `FoldersManager` encapsulates the logic for the `FoldersController` and uses `FolderManager` and `RoleManager` to implement it. This class will also use and return the future pydantic models.
But yeah... I agree the name is not the best... 😅 

Do you think these two should go together?

davelopez replied:
@jmchilton 
Do you prefer [this approach](https://github.com/galaxyproject/galaxy/pull/11686/files) to deal with the class encapsulating the controller logic instead of calling it a `Manager`? Or am I going completely offroad here? 😆 

Appreciated any advice on how to proceed with this :)

davelopez replied:
I have renamed the `FoldersManager` to `FoldersService` to better differentiate the class encapsulating the API actions logic from the regular Managers used to deal with the database models.
I have also used this naming in #11712 to unify it a bit. Again, the idea is to have this layer that directly speaks "pydantic" and uses the appropriate managers under the hood to expose the API actions that can be then called from the API controllers (either legacy or FastAPI) in a one-liner to reuse as much logic as possible and keed the API controllers thin.

But I am really open to better naming ideas or changes in this approach :)

davelopez committed:
Remove unused populators in FoldersApiTestCase

davelopez committed:
Rename FoldersManager to FolderService

This is to better differentiate the class encapsulating the API actions logic from the regular Manager used to deal with the database models.
